apache-bookkeeper/src/main/java/com/baeldung/BookkeeperLedgerExample.java

```diff
@@ -0,0 +1,72 @@
+package com.baeldung;
+
+
+import java.nio.ByteBuffer;
+import java.util.Iterator;
+import org.apache.bookkeeper.client.api.BookKeeper;
+import org.apache.bookkeeper.client.api.DigestType;
+import org.apache.bookkeeper.client.api.LedgerEntries;
+import org.apache.bookkeeper.client.api.LedgerEntry;
+import org.apache.bookkeeper.client.api.ReadHandle;
+import org.apache.bookkeeper.client.api.WriteHandle;
+import org.apache.bookkeeper.conf.ClientConfiguration;
+
+
+public class BookkeeperLedgerExample {
+
+  public static void main(String[] args) throws Exception {
+
```

Comment:
I think a better way to format this code would be to create your objects and have some functional methods within `BookkeeperLedger` and then have some test code that verifies the contents of the objects. Typically, we format the code for our tutorials using that method. There are lots of examples of this pattern within the `core-java` module.